plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSnapshots.java

```diff
@@ -74,4 +111,20 @@ private List<Long> getSnapshotsIdsByCreationOrder(String tableName)
                 .map(row -> (Long) row.getField(ID_FIELD))
                 .collect(toList());
     }
+
+    private Session sessionWithLegacySyntaxSupport()
+    {
+        return Session.builder(getSession())
+                .setCatalogSessionProperty("iceberg", "allow_legacy_snapshot_syntax", "true")
+                .build();
+    }
+
+    private String notSupportedError(String tableName, long id)
+    {
+        return format(
+                "Failed to access snapshot %s for table %s.\"%s\". This pattern for accessing Iceberg tables is not supported.",
```

Comment:
I think the message should mention a) the unsupported syntax that was used and b) direct users to use AS OF syntax, or enable compatibility toggle to temporarily restore previous behavior.